### ISSUE TITLE
`addOutputs` stripping underscores from names

### ISSUE THREAD

**Issue by justindra** [sst/sst#1863]
Not sure if this is intended or not. But just noticed that the stack output names do not match between what we provide and what is output.

```ts
  stack.addOutputs({
    API_URL: api.url,
    AUTH_CLIENT_ID: auth.cdk.userPoolClient.userPoolClientId,
    AUTH_DOMAIN: domain.domainName,
  });
```

outputs:
```
Stack justin-mws-Api
  Status: deployed
  Outputs:
    APIURL: ...
    AUTHCLIENTID: ...
    AUTHDOMAIN: ....
```

So it's stripping the `_` (underscores) out of the names.

**Comment by lefnire** [comment]
This is a nuisance indeed. Long story that I'd been putting outputs into an outputs file for CI/CD, and due to name-conflict with resource ID of... well, it's kinda hard to describe. Let's just say the solution is to move towards `sst.Config.Parameters`, but I can't do that just yet. 

@justindra I poked around and I think the underscore-stripping is done CDK-side. So I've got a feeling this is out of SST's jurisdiction. If it's a real issue where the outputs should match the key names put into `--outputs-file`, then we should be moving towards Config.Parameters instead.
